appsec/src/helper/subscriber/waf.cpp

```diff
@@ -236,6 +238,20 @@ void instance::listener::get_meta_and_metrics(
 {
     meta[std::string(tag::event_rules_version)] = ruleset_version_;
     metrics[tag::waf_duration] = total_runtime_;
+
+    for (const auto &[key, value] : schemas_) {
+        std::string schema = value;
+        if (value.length() > max_plain_schema_allowed) {
+            auto encoded = compress(schema);
+            if (encoded) {
+                schema = base64_encode(encoded.value(), false);
+            }
+        }
+
+        if (schema.length() <= max_schema_size) {
```

Comment:
I think it would be good to add a debug line here saying schema `key` has been discarded because of size.